test/e2e/functional_test.go

```diff
@@ -0,0 +1,368 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	watchtools "k8s.io/client-go/tools/watch"
+	"k8s.io/kubernetes/pkg/client/conditions"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const waitForPrometheusStartSeconds = 240
+
+var _ = g.Describe("[Feature:Prometheus][Conformance] Prometheus", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("prometheus")
+
+		url, bearerToken string
+	)
+
+	g.BeforeEach(func() {
+		var ok bool
+		url, bearerToken, ok = locatePrometheus(oc)
+		if !ok {
+			e2e.Skipf("Prometheus could not be located on this cluster, skipping prometheus test")
+		}
+	})
+
+	g.Describe("when run on the cluster", func() {
+
+		oc = exutil.NewCLI("monitoring", exutil.KubeConfigPath())
+
+		g.It("prometheus & alertmanager pods should use anti-affinity", func() {
```

Comment:
As mentioned in the comment, let me roughly walk you though how you could change this to use our e2e framework:

As this needs to check pods of prometheus and alertmanager, we would want these in their respective files.
So lets start with alertmanager:
- It would go in this file https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/alertmanager_test.go#L30
- To get the pod and check if the pod contains the value you have below, you would use something like:
`pods, err := f.KubeClient.CoreV1().Pods(f.Ns).List(metav1.ListOptions{FieldSelector: "status.phase=Running"})
`

that would get you the same podList as you get now, and it should be same as you do from now. This removes the need for exutil import.

Hopefully that makes sense? :)

Reply:
Of course , it has a lot of sense for me.
Thank you for your explanation, now it looks more clear and understandably. I will try to rewrite one case, and sorry in advance will asking stupid questions :).

Reply:
@wmuha No problem, feel free to ask!